wfs/wfs_service.py

```diff
@@ -42,38 +42,104 @@ def getTypenamesByServiceName(self, serviceName):
                 return {}
         return self.cachedTypenamesDict[serviceName]
 
+    def divideGeometry(self, geometry, parts):
+        """
+        Funkcja dzieląca geometrię na mniejsze części.
+        
+        :param geometry: Geometria, która ma zostać podzielona.
+        :param parts: Liczba części, na które ma zostać podzielona geometria.
+        :return: Lista mniejszych geometrii.
+        """
+        bounds = geometry.boundingBox()
+        width = bounds.width() / parts
+        height = bounds.height() / parts
+
+        sub_geometries = []
+        for i in range(parts):
+            for j in range(parts):
+                sub_bounds = QgsRectangle(
+                    bounds.xMinimum() + i * width,
+                    bounds.yMinimum() + j * height,
+                    bounds.xMinimum() + (i + 1) * width,
+                    bounds.yMinimum() + (j + 1) * height
+                )
+                sub_geometry = geometry.intersection(QgsGeometry.fromRect(sub_bounds))
+                if not sub_geometry.isEmpty():
+                    sub_geometries.append(sub_geometry)
+        
+        return sub_geometries
 
     def getWfsListbyLayer1992(self, layer, wfsService, typename):
+        """
+        Funkcja pobierająca dane z WFS, dzieląc geometrię na mniejsze części.
+        
+        :param layer: Warstwa, dla której wykonujemy zapytanie.
+        :param wfsService: Nazwa usługi WFS.
+        :param typename: Typ danych.
+        :return: Lista warstw z wynikami WFS.
+        """
         wfsUrl = WFS_URL_MAPPING[wfsService]
         wfsTypename = self.cachedTypenamesDict[wfsService][typename]
+        
+        # Dynamiczne zarządzanie tolerancją
+        area = layer.extent().area()
+
+        # Agregacja i uproszczenie geometrii
         aggregation = processing.run("native:aggregate",
-                       {'INPUT': layer,
-                        'GROUP_BY': 'NULL', 'AGGREGATES': [], 'OUTPUT': 'TEMPORARY_OUTPUT'})
+                    {'INPUT': layer,
+                    'GROUP_BY': 'NULL', 'AGGREGATES': [], 'OUTPUT': 'TEMPORARY_OUTPUT'})
+
         simplification = processing.run("native:simplifygeometries",
-                       {'INPUT': aggregation['OUTPUT'],
-                        'METHOD': 1, 'TOLERANCE': 100, 'OUTPUT': 'TEMPORARY_OUTPUT'})
+                    {'INPUT': aggregation['OUTPUT'],
+                    'METHOD': 1, 'TOLERANCE': 100, 'OUTPUT': 'TEMPORARY_OUTPUT'})
+
         simpleLayer = simplification['OUTPUT']
         feat = next(simpleLayer.getFeatures())
         geom = feat.geometry()
-        wkt = geom.asWkt()
-        wkt = roundCoordinatesOfWkt(wkt)    # zaokrąglanie współrzędnych do liczb całkowitych
+        geometry_type = feat.geometry().type()
 
+        # Jeżeli geometria nie jest punktowa, podziel ją na mniejsze fragmenty
+        geometries = [geom] if geometry_type == QgsWkbTypes.PointGeometry else self.divideGeometry(geom, 6)
 
```

Comment:
Ten fragment podczas testów znacząco obniżył działanie wtyczki w tej funkcjonalności. Lepiej będzie, jakby dzielenie warstwy było zrobione np. za pomocą narzędzia creategrid z natywnych poleceń qgisa, a wcześniej musiałyby być złączone obiekty bez duplikatów, tak jak to jest poniżej tego fragmentu